src/shared/sam/cli/samCliConfiguration.ts

```diff
@@ -55,7 +55,7 @@ export class DefaultSamCliConfiguration implements SamCliConfiguration {
         const detectedLocation = (await this._samCliLocationProvider.getLocation()) ?? ''
         // Avoid setting the value redundantly (could cause a loop because we
         // listen to the `onDidChangeConfiguration` event).
-        if (configLocation !== detectedLocation) {
+        if (detectedLocation && configLocation !== detectedLocation) {
```

Comment:
Looks like there are changes from another PR in here. Can you separate that PR out of this one now that this PR appears to have a clear scope?